tests/Microsoft.Identity.Test.Integration.netcore/WAM/RuntimeBrokerTests.cs

```diff
@@ -135,5 +136,100 @@ public async Task WamInteractiveAuthNoWindowHandleAsync()
+
+            AssertAuthResult(result, TokenSource.Broker, labResponse.Lab.TenantId);
+
+            // Get Accounts
+            var accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+            Assert.IsNotNull(accounts);
+
+            var account = accounts.FirstOrDefault();
+            Assert.IsNotNull(account);
+
+            // Acquire token silently
+            result = await pca.AcquireTokenSilent(scopes, account).ExecuteAsync().ConfigureAwait(false);
+
+            AssertAuthResult(result, TokenSource.Cache, labResponse.Lab.TenantId);
+
+            // Remove Account
+            await pca.RemoveAsync(account).ConfigureAwait(false);
+
+            // Assert the account is removed
+            accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(accounts);
+            Assert.AreEqual(0, accounts.Count());
+        }
+
+        [TestMethod]
+        public async Task WamUsernamePasswordRequestMsaPassthroughAsync()
+        {
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            string[] scopes = { "User.Read" };
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+               .Create("04f0c124-f2bc-4f59-8241-bf6df9866bbd")
+               .WithAuthority(labResponse.Lab.Authority, "organizations")
+               .WithWindowsBrokerOptions(new WindowsBrokerOptions()
+               {
+                   MsaPassthrough = true
+               })
+               .WithBrokerPreview().Build();
+
+            // Acquire token using username password
+            var result = await pca.AcquireTokenByUsernamePassword(scopes, labResponse.User.Upn, new NetworkCredential("", labResponse.User.GetOrFetchPassword()).SecurePassword).ExecuteAsync().ConfigureAwait(false);
+
+            AssertAuthResult(result, TokenSource.Broker, labResponse.Lab.TenantId);
+
+            // Get Accounts
+            var accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+            Assert.IsNotNull(accounts);
+
+            var account = accounts.FirstOrDefault();
+            Assert.IsNotNull(account);
+
+            // Acquire token silently
+            result = await pca.AcquireTokenSilent(scopes, account).ExecuteAsync().ConfigureAwait(false);
+
+            AssertAuthResult(result, TokenSource.Cache, labResponse.Lab.TenantId);
+
+            // Remove Account
+            await pca.RemoveAsync(account).ConfigureAwait(false);
+
+            // Assert the account is removed
+            accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(accounts);
+            Assert.AreEqual(0, accounts.Count());
+        }
+
+        private void AssertAuthResult(AuthenticationResult result, TokenSource tokenSource, string tenantId)
```

Comment:
Maybe this can move to the MsalAssert class to usable everywhere. seems like a useful assert. There is already a method there named AssertAuthResult so this will be a useful overload